nodestream/pipeline/extractors/credential_utils.py

```diff
@@ -51,13 +51,11 @@ def get_boto_session_with_refreshable_credentials(self):
         session._credentials = refreshable_credentials
         return session
 
-    def assume_role_if_supplied_and_get_session(self):
+    def make_client(self, client_name: str):
         if self.assume_role_arn:
-            return self.get_boto_session_with_refreshable_credentials()
-        return Session()
+            botocore_session = self.get_boto_session_with_refreshable_credentials()
+            return boto3.Session(
+                botocore_session=botocore_session, **self.session_args
+            ).client(client_name)
 
-    def make_client(self, client_name: str):
-        session = self.assume_role_if_supplied_and_get_session()
-        return boto3.Session(botocore_session=session, **self.session_args).client(
-            client_name
-        )
+        return boto3.client(client_name, **self.session_args)
```

Comment:
So this base client will refresh credentials using the user's arn by default?

Reply:
Good call. Turns out It will in some cases... but not all: https://boto3.amazonaws.com/v1/documentation/api/latest/guide/credentials.html 

I've updated it.